core/src/main/java/org/opensearch/sql/analysis/Analyzer.java

```diff
@@ -217,13 +220,35 @@ public LogicalPlan visitLimit(Limit node, AnalysisContext context) {
   public LogicalPlan visitFilter(Filter node, AnalysisContext context) {
     LogicalPlan child = node.getChild().get(0).accept(this, context);
     Expression condition = expressionAnalyzer.analyze(node.getCondition(), context);
+    verifySupportsCondition(condition);
 
     ExpressionReferenceOptimizer optimizer =
         new ExpressionReferenceOptimizer(expressionAnalyzer.getRepository(), child);
     Expression optimized = optimizer.optimize(condition, context);
     return new LogicalFilter(child, optimized);
   }
 
+  /**
+   * Ensure NESTED function is not used in WHERE clause. Fallback to legacy engine. Can remove when
+   * support is added for NESTED function in WHERE and GROUP BY clauses.
+   * @param condition : Filter condition
+   */
+  private void verifySupportsCondition(Expression condition) {
+    if (condition instanceof FunctionExpression) {
+      if (((FunctionExpression) condition).getFunctionName().getFunctionName().equalsIgnoreCase(
+          BuiltinFunctionName.NESTED.name()
+      )) {
+        throw new SyntaxCheckException(
```

Comment:
In pagination and JSON output we included a separate Error like https://github.com/opensearch-project/sql/pull/1497/files#diff-36239073fce7ec0b5f2d2f04c1eccec539d6d0f9b597734d2ca7cc3a881d3332

Reply:
I think it needs to be `SyntaxCheckException` for fallback functionality.